src/app/directives/restrict-special-char.directive.ts

```diff
@@ -0,0 +1,25 @@
+import { Directive, HostListener, Input } from '@angular/core';
+
+@Directive({
+  selector: '[appRestrictSpecialChar]'
+})
+export class RestrictSpecialCharDirective {
+  @Input() restrictedCharCode: any[];
+
+  constructor() { }
+
+  @HostListener('keypress') onkeypress(e) {
+    const event = e || window.event;
+    if (event) {
+      return this.isRestrictedChar(event);
+    }
+  }
+
+  isRestrictedChar(event) {
+    if (this.restrictedCharCode.length === 0) {
```

Comment:
this.restrictedCharCodes?.length

Reply:
what happens if we pass [restrictedCharCode]="" or [restrictedCharCode]="34, b"

Reply:
> what happens if we pass [restrictedCharCode]="" or [restrictedCharCode]="34, b"

`[restrictedCharCode]=""` will allow all characters but `[restrictedCharCode]="34, b"` will give error. If we use `[restrictedCharCode]="[34, b]"` instead of `[restrictedCharCode]="34, b"`, it will restrict entry of `"`.